src/dataset/dataset_utils.py

```diff
@@ -0,0 +1,43 @@
+import bisect
+from typing import List
+
+
+class VideosIndexer:
+    """
+    Handles absolute indexing for a list of videos.
+
+    Attributes:
+        video_list (List(list)) :matrix with the frame ids for each video
+        video_lengths (List(int)): partial sum array with lenghts of the videos
+    """
+    video_list: List[list]
+    video_lengths: List[int]
+
+    def __init__(self, video_list: List[List]):
```

Comment:
Nice usage of typing, thought it's better to always pass a type to the generic (the second `List` here is not parameterized). If you don't know this in advance, you can always use `typing.Any`, though I think in this case it should be `List[List[int]]`